pkg/operator/controller/certrotation_test.go

```diff
@@ -116,6 +147,8 @@ var _ = Describe("Cert rotation tests", func() {
 
 			checkCerts(client, namespace, false)
 
+			populateClientAndStoreWithSecrets(client, cm.(*certManager), namespace)
+			syncStoreWithLocalUpdateCalls(client, cm.(*certManager), namespace)
```

Comment:
@mhenriks looks like with this library-go change https://github.com/openshift/library-go/pull/1772 there's additional logic to call the CREATE.
not sure if this messes up functionality (should requeue on already exists err) but it definitely hurts the unit tests

I tried shifting the order a bit but it's clunky. Calling Sync() an extra time also doesn't help, seems like the lister is never updated and thus the library-go mechanism keeps trying to CREATE (which I guess makes sense since it's all fake)

Reply:
I got further here but can't help feeling like we're doing something fundamentally wrong because I don't ever recall being in this problem space in kubevirt
(client updates/patches needing to be reflected on the objects in the `cache.Store`)

Reply:
KubeVirt doesn't use this library